File: JFFliter/components/JFFliterItemCell.m
//
//  JFFliterItemCell.m
//  JFTX_WuHanPJJY
//
//  Created by geeksprite on 2017/7/21.
//  Copyright © 2017年 jf. All rights reserved.
//

#import <Masonry.h>
#import "JFFliterItemCell.h"
#import "JFFliterAppearanceManager.h"

@interface JFFliterItemCell ()
{
    UIButton *_chooseButton;
}
@property (nonatomic, strong) UIImageView *chooseImgView;

@end

@implementation JFFliterItemCell

#pragma mark - Init
- (instancetype)initWithFrame:(CGRect)frame {
    if (self = [super initWithFrame:frame]) {
        [self configureViews];
    }
    return self;
}

#pragma mark - Configure Views
- (void)configureViews {
    [self addSubview:self.chooseButton];
    [self.chooseButton mas_makeConstraints:^(MASConstraintMaker *make) {
        make.edges.equalTo(self);
    }];
    [self addSubview:self.chooseImgView];
    [self.chooseImgView mas_makeConstraints:^(MASConstraintMaker *make) {
        make.bottom.equalTo(self);
        make.right.equalTo(self);
        make.width.mas_equalTo(20);
        make.height.mas_equalTo(13);
    }];
}

#pragma mark - Getter && Setter
- (void)setChosen:(BOOL)chosen {
    _chosen = chosen;
    self.chooseButton.selected = self.chosen;
    self.chooseImgView.hidden = !chosen;
}

- (UIButton *)chooseButton {
    if (!_chooseButton) {
        _chooseButton = [UIButton buttonWithType:UIButtonTypeCustom];
        _chooseButton.titleLabel.font = [JFFliterAppearanceManager sharedManager].subButtonFont;
        _chooseButton.backgroundColor = [JFFliterAppearanceManager sharedManager].subButtonBackgroundColor;
        [_chooseButton setTitleColor:[JFFliterAppearanceManager sharedManager].subButtonTextChoodenColor forState:UIControlStateSelected];
        [_chooseButton setTitleColor:[JFFliterAppearanceManager sharedManager].subButtonTextNormalColor forState:UIControlStateNormal];
        _chooseButton.userInteractionEnabled = NO;
    }
    return _chooseButton;
}

- (UIImageView *)chooseImgView {
    if (!_chooseImgView) {
        _chooseImgView = [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"gouxuan"]];
    }
    return _chooseImgView;
}

@end

File: GGSwift/JFFliter/Components/JFFliterItemCell.swift
//
//  JFFliterItemCell.swift
//  GGSwift
//
//  Created by geeksprite on 2017/8/1.
//  Copyright © 2017年 聚风天下有限公司. All rights reserved.
//

import UIKit

class JFFliterItemCell: UICollectionViewCell {
    
    open lazy var chooseButton: UIButton = {
        let button = UIButton.init(type: .custom)
        button.titleLabel?.font = JFFliterAppearceManager.shared.subButtonFont
        button.backgroundColor = JFFliterAppearceManager.shared.subButtonBackgroundColor
        button.setTitleColor(JFFliterAppearceManager.shared.subButtonTextNormalColor, for: .normal)
        button.setTitleColor(JFFliterAppearceManager.shared.subButtonTextChoodenColor, for: .selected)
        button.isUserInteractionEnabled = false
        return button
    }()
    
    private lazy var chooseImgView: UIImageView = {
        let imgView = UIImageView()
        let path = Bundle.main.path(forResource: "gouxuan", ofType: "png")
        imgView.image = UIImage.init(contentsOfFile: path!)
        return imgView
    }()
    
    open var chosen: Bool? {
        didSet{
            self.chooseButton.isSelected = self.chosen!
            self.chooseImgView.isHidden = !(self.chosen!)
        }
    }
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        self.didInit()
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    
    override func layoutSubviews() {
        super.layoutSubviews()
        if self.width > 0.00001 {
            self.didLayoutSubviews()
        }
    }
    func didInit() {
        self.addSubview(self.chooseButton)
        self.addSubview(self.chooseImgView)
    }
    
    func didLayoutSubviews() {
        self.chooseButton.frame = self.bounds
        self.chooseImgView.frame = CGRect.init(x: self.width - 20.0, y: self.height - 13.0, width: 20.0, height: 13.0)
    }
}
